Add sound stuff + deps

sixtyfourktec opened the pull request.

jkloetzke reviewed (commented):
I'd like to put alsa-lib into `recipes/libs` because it's really just a library. All other sound stuff should go to the "multimedia" category. Especially pipewire because it's not just about sound... ;-)

sixtyfourktec commented on recipes/devel/lua.yaml:
```diff
@@ -0,0 +1,66 @@
+        packageScript: installPackageDev "$1/install/"
+        provideDeps: [ "*-dev" ]
+
+    tgt:
```
Done. I removed the tgt package, because lua actually doesn't provide any shared libs.

sixtyfourktec commented:
> I'd like to put alsa-lib into `recipes/libs` because it's really just a library. All other sound stuff should go to the "multimedia" category. Especially pipewire because it's not just about sound... ;-)

done

jkloetzke commented on recipes/libs/alsa-lib.yaml:
```diff
@@ -0,0 +1,24 @@
+
+metaEnvironment:
+    PKG_VERSION: "1.2.12"
+    PKG_LICENSE: "LGPL-2.1+ (library), GPL-2.0+ (aserver)"
```
This smells like we should split the package, shouldn't we?

jkloetzke replied:
Maybe we do it the other way around: explicitly only call `autotoolsPackageTgt` in the `packageScript`. Debian is not even packaging aserver. So I guess there is no real use of it anyway.

sixtyfourktec replied:
I removed that license, because actually I didn't build aserver.

jkloetzke commented on recipes/multimedia/pipewire.yaml:
```diff
@@ -0,0 +1,89 @@
+        provideDeps: [ "*-dev" ]
+
+    tgt:
+        packageScript: mesonPackageTgt
```
Would it make sense to split the package into the daemon with its modules and the library?

sixtyfourktec replied:
I did this and also for wireplumber.

jkloetzke commented on recipes/multimedia/pipewire.yaml:
```diff
@@ -0,0 +1,89 @@
+        -Dpipewire-alsa=enabled -Dalsa=enabled
+
+    mkdir -p install/.bob
+    cat >install/.bob/pipewire.user-table <<EOF
```
Again, we should have the infrastructure in place first.

sixtyfourktec commented:
I did redo the virt::udev package, because I believe having the general deps in there was wrong. Please have another look.

jkloetzke commented:
> I did redo the virt::udev package, because I believe having the general deps in there was wrong. Please have another look.

Yeah, that is certainly better. I added a fixup commit because I think this could be improved by removing duplicate stuff. There was also a `use: []` missing. Please have a look again and squash the fixup if you're ok with it.

sixtyfourktec commented:
> > I did redo the virt::udev package, because I believe having the general deps in there was wrong. Please have another look.
> 
> Yeah, that is certainly better. I added a fixup commit because I think this could be improved by removing duplicate stuff. There was also a `use: []` missing. Please have a look again and squash the fixup if you're ok with it.

I squashed the working parts of your fixup.

jkloetzke commented:
Thanks!
